units/src/block.rs

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Block height and interval types.
+//!
+//! These types are thin wrappers around `u32`, no invariants implemented or implied.
+//!
+//! These are general types for abstracting over block heights, they are not designed to use with
+//! lock times. If you are creating lock times you should be using the
+//! [`locktime::absolute::Height`] and [`locktime::relative::Height`] types.
+//!
+//! The difference between these types and the locktime types is that these types are thin wrappers
+//! whereas the locktime types contain more complex locktime specific abstractions.
+
+use core::{fmt, ops};
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+#[cfg(doc)]
+use crate::locktime;
+use crate::locktime::{absolute, relative};
+
+/// The block height, zero denotes the genesis block.
+///
+/// This type is not meant for constructing height based timelocks, this is a general purpose block
+/// height abstraction. For locktimes please see [`locktime::absolute::Height`].
+///
+/// This is a thin wrapper around a `u32` that may take on all values of a `u32`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+// Public to try and make it really clear that there are no invariants.
+pub struct BlockHeight(pub u32);
+
+impl BlockHeight {
+    /// Block height 0, the genesis block.
+    pub const ZERO: Self = BlockHeight(0);
+
+    /// The minimum block height (0), the genesis block.
+    pub const MIN: Self = Self::ZERO;
+
+    /// The maximum block height.
+    pub const MAX: Self = BlockHeight(u32::MAX);
+
+    /// Creates a block height from a `u32`.
+    // Because From<u32> is not const.
+    pub const fn from_u32(inner: u32) -> Self { Self(inner) }
+
+    /// Returns block height as a `u32`.
+    // Because type inference doesn't always work using `Into`.
+    pub const fn to_u32(&self) -> u32 { self.0 }
```

Comment:
What's the point of these if the field is `pub` anyway?

Reply:
For people who want to have an explicit method which describes what conversion is happening.